bin/cliain/src/commands.rs

```diff
@@ -0,0 +1,238 @@
+
+        /// Validator lower bound
+        #[clap(long)]
+        minimal_validator_stake: u64,
+
+        /// Maximum number of nominators
+        #[clap(long)]
+        max_nominators_count: Option<u32>,
+
+        /// Maximum number of validators
+        #[clap(long)]
+        max_validators_count: Option<u32>,
+    },
+
+    /// Transfer funds via balances pallet
+    Transfer {
+        /// Number of tokens to send,
+        #[clap(long)]
+        amount_in_tokens: u64,
+
+        /// SS58 id of target account
+        #[clap(long)]
+        to_account: String,
+    },
+
+    /// Send new runtime (requires sudo account)
+    UpdateRuntime {
+        #[clap(long)]
+        /// Path to WASM file with runtime
+        runtime: String,
+    },
+
+    /// Call staking validate call for a given controller
+    Validate {
+        /// Validator commission percentage
+        #[clap(long)]
+        commission_percentage: u8,
+    },
+
+    /// Update vesting for the calling account.
+    Vest,
+
+    /// Update vesting on behalf of the given account.
+    VestOther {
+        /// Account seed for which vesting should be performed.
+        #[clap(long)]
+        vesting_account: String,
+    },
+
+    /// Transfer funds via balances pallet
+    VestedTransfer {
+        /// Number of tokens to send.
+        #[clap(long)]
+        amount_in_tokens: u64,
+
+        /// Seed of the target account.
+        #[clap(long)]
+        to_account: String,
+
+        /// How much balance (in rappens, not in tokens) should be unlocked per block.
+        #[clap(long)]
+        per_block: Balance,
+
+        /// Block number when unlocking should start.
+        #[clap(long)]
+        starting_block: BlockNumber,
+    },
+
+    /// Print debug info of storage
+    DebugStorage,
+
+    /// Deploys a new contract, returns its code hash and the AccountId of the instance
+    /// contract cannot already exist on-chain
+    /// API signature: https://polkadot.js.org/docs/substrate/extrinsics/#instantiatewithcodevalue-compactu128-gas_limit-compactu64-storage_deposit_limit-optioncompactu128-code-bytes-data-bytes-salt-bytes
+    ContractInstantiateWithCode(ContractInstantiateWithCode),
+
+    /// Uploads new code without instantiating a contract from it
+    /// https://polkadot.js.org/docs/substrate/extrinsics/#uploadcodecode-bytes-storage_deposit_limit-optioncompactu128
+    ContractUploadCode(ContractUploadCode),
+
+    ///  Instantiates a contract from a previously deployed wasm binary.
+    /// API signature: https://polkadot.js.org/docs/substrate/extrinsics/#instantiatevalue-compactu128-gas_limit-compactu64-storage_deposit_limit-optioncompactu128-code_hash-h256-data-bytes-salt-bytes
+    ContractInstantiate(ContractInstantiate),
+
+    /// Calls a contract
+    /// API signature: https://polkadot.js.org/docs/substrate/extrinsics/#calldest-multiaddress-value-compactu128-gas_limit-compactu64-storage_deposit_limit-optioncompactu128-data-bytes
```

Comment:
a nit, but afaik the norm is to separate brief description from the rest of docstring with a single line; iirc this will prevent terminal help from displaying these very long urls, which imho is an improvement as well

Reply:
fixed